python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,581 @@
+        result.sources = result.sources_footprints.asAstropy()
+
+        return result
+
+    def _find_sources(self, exposure, background, calib_sources, id_generator):
+        """Detect and measure sources on the exposure.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to detect and measure sources on; must have a valid PSF.
+        background : `lsst.afw.math.BackgroundList`
+            Background that was fit to the exposure during detection;
+            modified in-place during subsequent detection.
+        calib_sources : `astropy.table.Table`
+            Per-visit catalog of measurements to get 'calib_*' flags from.
+        id_generator : `lsst.meas.base.IdGenerator`
+            Object that generates source IDs and provides random seeds.
+
+        Returns
+        -------
+        sources
+            Catalog that was detected and measured on the exposure.
+        """
+        table = afwTable.SourceTable.make(self.sources_schema.schema, id_generator.make_table_id_factory())
+
+        self.repair.run(exposure=exposure)
+        detections = self.detection.run(table=table, exposure=exposure, background=background)
+        sources = detections.sources
+
+        self.sky_sources.run(exposure.mask, id_generator.catalog_id, sources)
+
+        self.deblend.run(exposure=exposure, sources=sources)
+        # The deblender may not produce a contiguous catalog; ensure
+        # contiguity for subsequent tasks.
+        if not sources.isContiguous():
+            sources = sources.copy(deep=True)
+
+        self.measurement.run(sources, exposure)
+        self.normalized_calibration_flux.run(exposure=exposure, catalog=sources)
+        self.apply_aperture_correction.run(sources, exposure.apCorrMap)
+        self.catalog_calculation.run(sources)
+        self.set_primary_flags.run(sources)
+
+        sources["visit"] = exposure.visitInfo.id
+        sources["detector"] = exposure.info.getDetector().getId()
+
+        self._match_calib_sources(sources, calib_sources, exposure.info.getDetector().getId())
+
+        return sources
+
+    def _match_calib_sources(self, sources, calib_sources, detector):
+        """Match with calib_sources to set `calib_*` flags in the output
+        catalog.
+
+        Parameters
+        ----------
+        sources : `lsst.afw.table.SourceCatalog`
+            Catalog that was detected and measured on the exposure. Modified
+            in place to set the psf_fields.
+        calib_sources : `astropy.table.Table`
+            Per-visit catalog of measurements to get 'calib_*' flags from.
+        detector : `int`
+            Id of detector for this exposure, to get the correct sources from
+            calib_sources for cross-matching.
+        """
+        # NOTE: we don't remove the sky sources here, but they should be very
+        # far from any actual source, and so should not match with anything.
```

Comment:
This makes me a little nervous just because the catalogs are measured slightly differently, but I do agree it would be very pathological for a sky source to accidentally match a psf source.

Reply:
Trying to doing a multi-level selection with an afw catalog turned out to be such a pain that I decided to just skip it. If it turns out to be a problem (I'm also slightly nervous about it, but agree that you'd have to have a very strange case to match a sky source to a calib star within 0.2"!) someone in the future can be mad at me.